Sprint 16

Depechist opened the pull request.

Depechist committed:
Экран Онбординга

- Добавлен экран Онбординга согласно дизайну в Фигме

Depechist committed:
Создание категории

- Созданы экраны создания Категории
- Добавлена логика создания и хранения Категории

amardanov commented on Tracker/Onboarding/OnboardingPageViewController.swift:
```diff
@@ -0,0 +1,203 @@
+    }
+    
+    // Создаем стоковую страницу Онбординга с изображением и текстом
+    func createOnboardingPage(imageName: String, labelText: String) -> UIViewController {
```
`private`

amardanov commented on Tracker/SceneDelegate.swift:
```diff
@@ -18,28 +18,31 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+        // Создаем экран Онбординга
+        let onboardingVC = OnboardingPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
+        
+//        // Создаем и добавляем экран Трекеров в TabBar
```
Закомментированный код лучше не оставлять в коде приложения. Так код будет чище и ревьюеру будет проще проверять за счет уменьшения количества строк

amardanov commented on Tracker/Onboarding/OnboardingPageViewController.swift:
```diff
@@ -0,0 +1,203 @@
+    // MARK: - Initializers
+    
+    // Настройка переходов между страницами
+    override init(transitionStyle style: UIPageViewController.TransitionStyle, navigationOrientation: UIPageViewController.NavigationOrientation, options: [UIPageViewController.OptionsKey : Any]? = nil) {
```
Рекомендую придерживаться следующего стиля написания длинных вызовов и длинных объявлений функций. Как пример можешь использовать то, что я напишу. Такой формат принят за основу во многих компаниях и следование ему даст несколько очков к карме при работе над боевыми проектами

```
override init(
        transitionStyle style: UIPageViewController.TransitionStyle,
        navigationOrientation: UIPageViewController.NavigationOrientation,
        options: [UIPageViewController.OptionsKey : Any]? = nil
    ) {
```

Depechist committed:
Онбординг. Исправления

- Исправлен PageControl. Теперь отображается корректно
- Запоминается нажатие кнопки на Онбординга. Теперь Онбординг не показывается тем пользователям, которые его уже видели ранее и нажали на кнопку "Вот это технологии!".

Depechist committed:
Статистика + Темная тема

В работе.

- Создан экран статистики с логикой. Баг: экран всегда показывает коллекцию, будет исправлено.
- Добавлена темная тема

Depechist committed:
Аналитика

- Добавлена Яндекс.Метрика и События на следующие эвенты:

- Открытие
- Закрытие
- Тап на добавление трекера
- Тап на кнопку о выполнении трекера

Depechist committed:
Статистика + Темная тема. Исправления

- Исправлен экран статистики. Теперь он корректно реагирует на количество трекеров или их отсутствие
- Статистика теперь корректно отображает количество выполненных трекеров
- Исправлен баг с цветом подложки таблицы на экране Статистики
- Таблицу на экране Статистики теперь нельзя скроллить
- Исправлен баг с некорректным отображением плейсхолдера на экране Статистики
- Исправлено поведение экрана Онбординга. Теперь элементы не меняют свой цвет, если включена темная тема - согласно дизайну в Фигме.

Depechist committed:
Исправления

- Исправлена "прозрачность" tabBar (исправлен нижний констрейнт у коллекции), а также добавлена полоска сверху согласно дизайну в Фигме
- В TrackersViewController добавлены плейсхолдеры для пустой выдачи в поиске, необходимо дописать логику под корректное отображение

Depechist committed:
Скриншот-тест

- Добавлен скриншот-тест, который успешно пройден

Depechist committed:
Контекстное меню + другое

- Добавлено контекстное меню с функционалом Закрепа, Редактирования и Удаления трекера
- Добавлена аналитика на События по действиям в контекстном меню
- На главный экран добавлена кнопка Фильтры
- Добавлена аналитика на тапы по кнопке Фильтры
- Дописана логика отображения плейсхолдера emptySearch в случае пустой коллекции при поиске Трекеров

Depechist committed:
Локализация

- Добавлены локализации: русский язык, английский язык
